test/ehttpc_tests.erl

```diff
@@ -690,3 +690,76 @@ dedup(L) ->
 dedup(H, []) -> [H];
 dedup(H, [H | T]) -> dedup(H, T);
 dedup(H, [X | T]) -> [H | dedup(X, T)].
+
+%% Test case to check that requests are finished as they should be even when
+%% there is no body and headers indicate that there should be a body. The gun
+%% library handles this in a special way. See
+%% https://ninenines.eu/docs/en/gun/1.3/manual/gun.data/ and
+%% https://github.com/ninenines/gun/issues/141
+no_body_but_headers_indicating_body_test_() ->
+    snabbkaffe:fix_ct_logging(),
+    Port = ?PORT,
+    ServerOpts = #{port => Port, name => ?FUNCTION_NAME, oneoff => false},
+    PoolOpts = pool_opts(Port, false),
+    RequestsWithBody =
+        fun() ->
+            [
+                begin
+                    Response1 = ehttpc:request(
+                        ?POOL,
+                        RequestType,
+                        {<<"/">>, [{<<"content-type">>, <<"text/html">>}], <<>>},
+                        5000,
+                        0
+                    ),
+                    ok = ensure_not_error_response(Response1),
+                    Response2 = ehttpc:request(
+                        ?POOL,
+                        RequestType,
+                        {<<"/">>, [{<<"content-length">>, <<"0">>}], <<>>},
+                        5000,
+                        0
+                    ),
+                    ok = ensure_not_error_response(Response2)
+                end
+             || RequestType <- [put, post]
```

Comment:
Q: should this also repeat the `put` at the end so that `post` also has a request after it?  Similar to the note below.

Reply:
`RequestsWithNoBody()` will be executed afterwards so there will be requests after the post request anyway.